teps/0029-step-workspaces.md

```diff
@@ -57,6 +69,99 @@ that has access to those contents as well.
 - A Task author can still use the volume "hack" to attach `Workspaces` to `Sidecars` in
 combination with the feature proposed here.
 
+## Proposal
+
+Add `workspaces` lists to Steps and Sidecars. Here's the existing YAML and behaviour:
+
+```yaml
+spec:
+  workspaces:
+  - name: foo
+  steps:
+  - image: ubuntu
+    script: # ... This Step automatically mounts "foo" workspace.
+  - image: ubuntu
+    script: # ... And so does this Step.
+```
+
+And here's a Task using the new Step Workspaces feature:
+
+```yaml
+spec:
+  workspaces:
+  - name: foo
+  steps:
+  - image: ubuntu
+    workspaces:
+    - name: foo
+    script: # ... This Step mounts "foo" workspace.
+  - image: ubuntu
+    script: # ... But this Step does not.
+```
+
+The existing YAML and behaviour will continue to be supported.
+
+### Add `workspaces` to `Steps`
+
+1. Add a `workspaces` list to `Steps`.
+2. Allow `workspaces` from the `Task` to be explicitly named in that list like this:
+
+    ```yaml
+    workspaces:
+    - name: my-sensitive-workspace
+    steps:
+    - name: foo
+      workspaces:
+      - name: my-sensitive-workspace
+    ```
+3. When a `workspace` is listed in a Step, it is no longer automatically mounted - either to
+`Steps` or `Sidecars` - unless they also have the `workspace` in their own `workspaces` list.
+
+Example YAML:
+
+```yaml
+# task spec
+spec:
+  workspaces:
+  - name: git-ssh-credentials
+    mountPath: /root/.ssh
+  steps:
+  - name: clone-repo
+    image: alpine/git:v2.26.2
+    workspaces:
+    - name: git-ssh-credentials
```

Comment:
Could we specify the `mountPath` here (in addition to above ?)

Reply:
Main reason is if we want to use the workspace in two steps (but not the other ones) but not mounted at the same path.

Reply:
I'm definitely open to adding that feature.  Sure thing, will update TEP to include it now.

Reply:
Updated with addition to the proposal to support this.